src/main/java/lotto/LottoShop.java

```diff
@@ -19,33 +20,44 @@ public LottoShop(LottoBundle lottoBundle) {
         this.lottoBundle = lottoBundle;
     }
 
-    public void buyAuto(int price) {
-        validatePrice(price);
-        int lottoCount = Math.floorDiv(price, PRICE_PER_PIECE);
-        lottoBundle = new LottoBundle(toList(lottoCount));
+    public void buyLotto(LottoRequestDto lottoRequestDto) {
+        Purchase purchase = new Purchase(lottoRequestDto.getAmount(), lottoRequestDto.getManualCount());
+
+        lottoBundle = buyAuto(purchase.countOfAuto());
+        if (existManulLotto(lottoRequestDto)) {
+            List<String> manualLottoStrings = lottoRequestDto.getManualLottoStrings();
+            lottoBundle = lottoBundle.join(buyManual(manualLottoStrings));
+        }
```

Comment:
```suggestion
    public void buy(LottoRequestDto lottoRequestDto) {
        Purchase purchase = new Purchase(lottoRequestDto.getAmount(), lottoRequestDto.getManualCount());

        this.lottoBundle = buy(lottoRequestDto, purchase);
    }

    private LottoBundle buy(LottoRequestDto lottoRequestDto, Purchase purchase) {
        LottoBundle lottoBundle = buyAuto(purchase.countOfAuto());
        if (purchase.isManual()) {
            LottoBundle manualLotto = buyManual(lottoRequestDto.getManualLottoNumbers());
            return lottoBundle.join(manualLotto);
        }
        return lottoBundle;
    }
```

DTO의 값을 계속 꺼내서 활용하기 보다는, 이미 생성한 객체를 활용하는 것은 어떨까요